main.ts

```diff
@@ -72,6 +76,44 @@ class Translator {
     this.visit(expr);
   }
 
+  /** Searches for super() calls and emits an initializer for it if found. */
+  maybeEmitInitializerForSuperCall(body: ts.Block) {
+    if (!body || body.statements.length === 0) return;
+
+    var findSuperCalls = (stmt: ts.Statement) => {
+      if (stmt.kind !== ts.SyntaxKind.ExpressionStatement) return;
+      var nestedExpr = (<ts.ExpressionStatement>stmt).expression;
+      if (nestedExpr.kind !== ts.SyntaxKind.CallExpression) return;
+      var callExpr = <ts.CallExpression>nestedExpr;
+      if (callExpr.expression.kind !== ts.SyntaxKind.SuperKeyword) return;
+
+      this.emit(': super (');
+      this.visitList(callExpr.arguments);
+      this.emit(')');
+    };
+
+    body.statements.forEach(findSuperCalls);
+  }
+
+  /**
+   * Checks whether `callExpr` is a super() call that should be ignored because it was already
+   * handled by `maybeEmitSuperInitializer` above.
+   */
+  isHandledSuperCall(callExpr: ts.CallExpression): boolean {
```

Comment:
does this make it easy for the two methods to get out-of-sync? like we make a change above so some expression is now a handled super call, but forget to update this method to also return true for that case?
Maybe instead this could return the expressions discovered in the ctorBody which should be produced as initializers?